src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -79,9 +81,23 @@ protected MtGetRecordsResult getAllRecords(GetRecordsRequest request, StreamArn
             // when called to map a record, extract tenant context and table
             final FieldValue<?> fieldValue = fieldValueFunction.apply(record.getDynamodb().getKeys());
             // then establish context to map physical to virtual record, i.e., remove index key prefixes
-            return mtDynamoDb.getMtContext().withContext(fieldValue.getContext(), () ->
-                mtDynamoDb.getTableMapping(fieldValue.getTableName()).getRecordMapper().apply(record)
-            );
+            return mtDynamoDb.getMtContext().withContext(fieldValue.getContext(), () -> {
+                try {
+                    return mtDynamoDb.getTableMapping(fieldValue.getTableName()).getRecordMapper().apply(record);
+
+                } catch (UncheckedExecutionException e) {
```

Comment:
IMHO, it would be better to either throw a `ResourceNotFoundException` from `getTableMapping` (i.e., unwrap inside the method) or return `Optional`. Otherwise the implementation leaks out.